### ISSUE TITLE
Publish / Subscribe API 

### ISSUE THREAD

**Issue by kilianc** [kilianc/adobe-generator#10]
The following new _events_ should be published from photoshop following this table:

<table width="100%">
    <thead>
        <tr>
            <th>Event Name</th>
            <th>Event Trigger</th>          
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>layer.new</td>
            <td>Triggered when a new layer is created</td>
        </tr>
        <tr>
            <td>layer.rename</td>
            <td>Triggered when a new layer is renamed</td>
        </tr>
        <tr>
            <td>layer.delete</td>
            <td>Triggered when a new layer is deleted</td>
        </tr>
        <tr>
            <td>layer.contentChange</td>
            <td>
                Triggered when a the content of the layer changes, this could be at pixel level for a 
                bitmap layer, the solid color of a shape, a new mask, applied, adjustment applied,
                blending mode changed or some transformation occurred.
            </td>
        </tr>
        <tr>
            <td>layer.move</td>
            <td>
                Triggered when a layer changes is X,Y
                (the assumption is that the origin is on the top-left and unit is pixel)
            </td>
        </tr>
        <tr>
            <td>layer.zOrder</td>
            <td>Triggered when a layer changes his position inside the PS layer stack</td>
        </tr>
        <tr>
            <td>layer.visibility</td>
            <td>Triggered when a layer is hidden/unhidden</td>
        </tr>
        <tr>
            <td>layer.opacity</td>
            <td>Triggered when a layer is locked/unlocked</td>
        </tr>
        <tr>
            <td>layer.fill</td>
            <td>Triggered when a layer is locked/unlocked</td>
        </tr>
        <tr>
            <td>layer.lock</td>
            <td>Triggered when a layer is locked/unlocked</td>
        </tr>
        <tr>
            <td>doc.close</td>
            <td>Triggered when the document is closed</td>
        </tr>
        <tr>
            <td>doc.save</td>
            <td>Triggered when the document is saved to disk/cloud</td>
        </tr>
        <tr>
            <td>doc.apocalypse</td>
            <td>
                Triggered when something that affects every layer happens,
                for instance a canvas resize, an image resize, a crop tool etc...
            </td>
        </tr>
    </tbody>
</table>

Each subscription to these events should accept an optional JS script.
This script should be executed every time the event triggers and his output carried as response.

Basically each new event will carry a JSON encoded string like the following:

``` js
{
    document_id: Number,
    layer_id: Number,
    event_name: String,
    script_output: Object
}
```

**Comment by kilianc** [comment]
@TomRuark I just added fill and opacity, I think that these 2 events should be a part of those "easy to implement". Can you confirm that?
